File: ReviewService/models/review.go
package models

import "time"

type Review struct {
	Id        int64      `json:"id"`
	UserId    int64      `json:"user_id"`
	BookingId int64      `json:"booking_id"`
	HotelId   int64      `json:"hotel_id"`
	Comment   string     `json:"comment"`
	Rating    int        `json:"rating"`
	CreatedAt time.Time  `json:"created_at"`
	UpdatedAt time.Time  `json:"updated_at"`
	DeletedAt *time.Time `json:"deleted_at"`
	IsSynced  bool       `json:"is_synced"`
}

File: e2e-tests/api-flows/clients/review.ts
const BASE = process.env.REVIEW_SERVICE_URL || 'http://localhost:3004';

function headers(userId: string, email: string, role: string = 'customer'): Record<string, string> {
  return {
    'Content-Type': 'application/json',
    'X-User-ID': userId,
    'X-User-Email': email,
    'X-User-Role': role,
  };
}

export async function createReview(
  data: { user_id: number; booking_id: number; hotel_id: number; comment: string; rating: number },
  userId: string,
  email: string,
) {
  const res = await fetch(`${BASE}/reviews/`, {
    method: 'POST',
    headers: headers(userId, email),
    body: JSON.stringify(data),
  });
  const body = await res.json();
  return { status: res.status, data: body };
}

export async function getReviewsByBookingId(bookingId: number) {
  const res = await fetch(`${BASE}/reviews/booking/${bookingId}`, {
    headers: { 'Content-Type': 'application/json' },
  });
  const body = await res.json();
  return { status: res.status, data: body };
}

export async function getReviewsByHotelId(hotelId: number) {
  const res = await fetch(`${BASE}/reviews/hotels/${hotelId}`, {
    headers: { 'Content-Type': 'application/json' },
  });
  const body = await res.json();
  return { status: res.status, data: body };
}

export async function healthCheck() {
  const res = await fetch(`${BASE}/health`);
  return res.ok;
}
